src/textfont.ts

```diff
@@ -9,33 +9,93 @@
 import { Vex } from './vex';
 import { PetalumaScriptTextMetrics } from './fonts/petalumascript_textmetrics';
 import { RobotoSlabTextMetrics } from './fonts/robotoslab_textmetrics';
+export interface TextFontMetrics {
+  advanceWidth: number;
+  ha: number;
+}
+
+export interface TextFontGlyph {
+  x_min: number;
+  x_max: number;
+  y_min: number;
+  y_max: number;
+  ha: number;
+  leftSideBearing: number;
+  advanceWidth: number;
+}
+
+export interface TextFontRegistry {
+  [name: string]: unknown;
+  name?: string;
+  resolution?: number;
+  glyphs?: Record<string, TextFontGlyph>;
+  family: string;
+  serifs: boolean;
+  monospaced?: boolean;
+  italic: boolean;
+  bold: boolean;
+  maxSizeGlyph?: string;
+  superscriptOffset?: number;
+  subscriptOffset?: number;
+  description: string;
+}
+
+export interface TextFontAttributes {
+  type: string;
+}
 
 // To enable logging for this class. Set `Vex.Flow.TextFont.DEBUG` to `true`.
-function L(...args) {
+function L(
+  // eslint-disable-next-line
+  ...args: any[]) {
   if (TextFont.DEBUG) Vex.L('Vex.Flow.TextFont', args);
 }
 
 export class TextFont {
-  static get CATEGORY() {
+  protected static debug: boolean;
+  protected resolution: number = 1000;
+  protected name?: string;
+  protected glyphs: Record<string, TextFontMetrics> = {};
+  protected family: string = '';
+  protected serifs?: boolean;
+  protected monospaced?: boolean;
+  protected italic?: boolean;
+  protected bold?: boolean;
+  protected superscriptOffset?: number;
+  protected subscriptOffset?: number;
+  protected description?: string;
+  protected maxSizeGlyph: string;
+  protected weight: string;
+  protected style: string;
+  protected fontCacheKey: string = '';
+
+  protected static registryInstance: TextFontRegistry[];
+  protected static textWidthCacheInstance?: Record<string, Record<string, number>>;
+
+  protected size: number;
+  protected attrs: TextFontAttributes;
+
+  static get CATEGORY(): string {
     return 'textFont';
   }
 
-  static get DEBUG() {
+  static get DEBUG(): boolean {
     return TextFont.debug;
   }
-  static set DEBUG(val) {
+
+  static set DEBUG(val: boolean) {
     TextFont.debug = val;
   }
 
   // ### fontRegistry
   // Getter of an array of available fonts.  Applications may register their
   // own fonts and the metrics for those fonts will be available to the
   // application.
-  static get fontRegistry() {
+  static get fontRegistry(): TextFontRegistry[] {
     if (!TextFont.registryInstance) {
       TextFont.registryInstance = [];
       TextFont.registryInstance.push({
-        name: 'Roboto Slab',
+        name: 'RobotoSlab',
```

Comment:
Is this intentional? (The space is missing.)

Reply:
Hi, the space is in the font name, probably a mistake:

`<link href="https://fonts.googleapis.com/css2?family=Roboto+Slab:wght@500&display=swap" rel="stylesheet">`

Reply:
Not really. Reversed.